src/Build.UnitTests/Graph/GraphTestingUtilities.cs

```diff
@@ -0,0 +1,400 @@
+        }
+
+        public static SolutionFileBuilder FromGraph(
+            ProjectGraph graph,
+            Dictionary<string, Dictionary<SolutionConfigurationInSolution, ProjectConfigurationInSolution>> projectConfigurations = null,
+            (string, string)[] solutionDependencies = null,
+            (string, string[])[] solutionDependenciesProjectNameToGuids = null)
+        {
+            return new SolutionFileBuilder
+            {
+                Projects = graph.ProjectNodes.ToDictionary(
+                    n => GraphTestingUtilities.GetProjectNumber(n).ToString(),
+                    n => n.ProjectInstance.FullPath),
+                ProjectConfigurations = projectConfigurations,
+                SolutionDependencies = solutionDependencies,
+                SolutionDependenciesProjectNameToGuids = solutionDependenciesProjectNameToGuids
+            };
+        }
+
+        public string BuildSolution()
+        {
+            var projectInfos = Projects.ToDictionary(
+                kvp => kvp.Key,
+                kvp => new ProjectInfo(
+                    kvp.Key,
+                    kvp.Value,
+                    "{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}",
+                    Guid.NewGuid()
+                        .ToString("B")));
+
+            if (SolutionDependencies != null && SolutionDependencies.Length > 0)
+            {
+                foreach (var (parent, dependency) in SolutionDependencies)
+                {
+                    projectInfos.ShouldContainKey(parent);
+                    projectInfos.ShouldContainKey(dependency);
+
+                    projectInfos[parent].AddDependency(projectInfos[dependency]); 
+                }
+            }
+
+            if (SolutionDependenciesProjectNameToGuids!= null && SolutionDependenciesProjectNameToGuids.Length > 0)
+            {
+                foreach (var (parent, dependencyGuids) in SolutionDependenciesProjectNameToGuids)
+                {
+                    projectInfos.ShouldContainKey(parent);
+                    projectInfos[parent].AddGuidDependencies(dependencyGuids);
+                }
+            }
+
+            ProjectConfigurations = ProjectConfigurations ?? projectInfos.ToDictionary(
+                p => p.Key,
+                p => SolutionConfigurationPlatforms.ToDictionary(
+                    sc => sc,
+                    sc => new ProjectConfigurationInSolution(
+                        sc.ConfigurationName,
+                        sc.PlatformName,
+                        true)
+                    )
+                );
+
+            var sb = new StringBuilder();
+
+            sb.AppendLine($@"
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version {VisualStudioVersion.VersionLatest}
+VisualStudioVersion = {VisualStudioVersion.VersionLatest}.0.0.0
```

Comment:
Since the content of the file is fairly fixed I think it'd be more accurate to pin this at the version that you took the format from.